packages/adapter/src/utils.ts

```diff
@@ -175,3 +175,15 @@ export function getDecimalBN(decimals: number | string) {
   const power = new BN(decimals, 10);
   return base.pow(power);
 }
+
+/**
+ * Determine the account type based on the address.
+ * Because 1 byte takes 2 characters in hex string, we can use the length of the address to determine the address type.
+ * AccountKey20: 42 characters with prefix 0x.
+ * AccountId32: 66 characters with prefix 0x.
```

Comment:
No substrate wallet address contains "0x", so AccountId32 length check should be 64. This function needs to throw an error on all wrong cases instead of using AccountType.AccountId32 as the fall-back.

For example,
```
if starts with 0x
   if length is 42
       AccountType.AccountKey20
else
   if length is 40
       AccountType.AccountKey20
   if length is 64
       AccountType.AccountId32

throw new Error(`Unrecognized address format`);

```

Reply:
Fixed